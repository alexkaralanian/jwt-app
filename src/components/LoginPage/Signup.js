import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import "./LoginPage.css";

export default function Signup(props) {
  return (
    <div className="container ">
      <div className="centeredContainer">
        <div className="borderBox ">
          <div className="loginFormHeader">
            <div className="loginIcon">
              <i className="fa fa-shield fa-5x" />
            </div>
          </div>

          <form id="loginform" onSubmit={props.handleSubmit}>
            <div className="form-group">
              <div className="input-group">
                <div className="input-group-addon">
                  <i className="fa fa-user" />
                </div>
                <input
                  name="firstName"
                  type="text"
                  className="form-control"
                  placeholder="First Name"
                  value={props.firstName}
                  onChange={props.handleChange}
                />
              </div>
            </div>

            <div className="form-group">
              <div className="input-group">
                <div className="input-group-addon">
                  <i className="fa fa-user" />
                </div>
                <input
                  name="lastName"
                  type="text"
                  className="form-control"
                  placeholder="Last Name"
                  value={props.lastName}
                  onChange={props.handleChange}
                />
              </div>
            </div>

            <div className="form-group">
              <div className="input-group">
                <div className="input-group-addon">
                  <i className="fa fa-envelope" />
                </div>
                <input
                  name="email"
                  type="text"
                  className="form-control"
                  placeholder="Email"
                  value={props.email}
                  onChange={props.handleChange}
                />
              </div>
            </div>

            <div className="form-group">
              <div className="input-group">
                <div className="input-group-addon">
                  <i className="fa fa-lock" />
                </div>
                <input
                  name="password"
                  type="password"
                  className="form-control"
                  placeholder="Password"
                  value={props.password}
                  onChange={props.handleChange}
                />
              </div>
            </div>
            <div className="buttonGroup">
              <div className="form-group buttonGroup">
                <input
                  type="submit"
                  className="signup btn btn-primary"
                  value="Signup"
                />
              </div>
            </div>
          </form>

          <div>
            <p className="signupRedirect">
              <Link to="/login">Have an account? Login Here</Link>
            </p>
          </div>

          <div>
            {props.error ? <p className="error">{props.error}</p> : null}
          </div>
        </div>
      </div>
    </div>
  );
}

Signup.defaultProps = {
  email: "",
  password: "",
  error: "",
  handleChange: () => {},
  handleSubmit: () => {}
};

Signup.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  email: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  // isFetching: PropTypes.bool.isRequired,
  // onAuth: PropTypes.func.isRequired,
  error: PropTypes.string.isRequired
};
