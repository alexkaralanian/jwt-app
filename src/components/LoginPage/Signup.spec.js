import { MemoryRouter } from "react-router-dom";
import React from "react";
import { shallow, mount } from "enzyme";
import Signup from "./Signup";

test("Signup renders without crashing", () => {
  mount(
    <MemoryRouter>
      <Signup />
    </MemoryRouter>
  );
});

test(" Signup Component renders correctly", () => {
  const component = shallow(<Signup />);
  expect(component).toMatchSnapshot();
});
