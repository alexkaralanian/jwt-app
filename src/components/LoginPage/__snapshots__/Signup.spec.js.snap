// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Signup Component renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="container "
>
    <div
        className="centeredContainer"
    >
        <div
            className="borderBox "
        >
            <div
                className="loginFormHeader"
            >
                <div
                    className="loginIcon"
                >
                    <i
                        className="fa fa-shield fa-5x"
                    />
                </div>
            </div>
            <form
                id="loginform"
                onSubmit={[Function]}
            >
                <div
                    className="form-group"
                >
                    <div
                        className="input-group"
                    >
                        <div
                            className="input-group-addon"
                        >
                            <i
                                className="fa fa-user"
                            />
                        </div>
                        <input
                            className="form-control"
                            name="firstName"
                            onChange={[Function]}
                            placeholder="First Name"
                            type="text"
                            value={undefined}
                        />
                    </div>
                </div>
                <div
                    className="form-group"
                >
                    <div
                        className="input-group"
                    >
                        <div
                            className="input-group-addon"
                        >
                            <i
                                className="fa fa-user"
                            />
                        </div>
                        <input
                            className="form-control"
                            name="lastName"
                            onChange={[Function]}
                            placeholder="Last Name"
                            type="text"
                            value={undefined}
                        />
                    </div>
                </div>
                <div
                    className="form-group"
                >
                    <div
                        className="input-group"
                    >
                        <div
                            className="input-group-addon"
                        >
                            <i
                                className="fa fa-envelope"
                            />
                        </div>
                        <input
                            className="form-control"
                            name="email"
                            onChange={[Function]}
                            placeholder="Email"
                            type="text"
                            value=""
                        />
                    </div>
                </div>
                <div
                    className="form-group"
                >
                    <div
                        className="input-group"
                    >
                        <div
                            className="input-group-addon"
                        >
                            <i
                                className="fa fa-lock"
                            />
                        </div>
                        <input
                            className="form-control"
                            name="password"
                            onChange={[Function]}
                            placeholder="Password"
                            type="password"
                            value=""
                        />
                    </div>
                </div>
                <div
                    className="buttonGroup"
                >
                    <div
                        className="form-group buttonGroup"
                    >
                        <input
                            className="signup btn btn-primary"
                            type="submit"
                            value="Signup"
                        />
                    </div>
                </div>
            </form>
            <div>
                <p
                    className="signupRedirect"
                >
                    <Link
                        replace={false}
                        to="/login"
                    >
                        Have an account? Login Here
                    </Link>
                </p>
            </div>
            <div />
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className="container "
>
      <div
            className="centeredContainer"
      >
            <div
                  className="borderBox "
            >
                  <div
                        className="loginFormHeader"
                  >
                        <div
                              className="loginIcon"
                        >
                              <i
                                    className="fa fa-shield fa-5x"
                              />
                        </div>
                  </div>
                  <form
                        id="loginform"
                        onSubmit={[Function]}
                  >
                        <div
                              className="form-group"
                        >
                              <div
                                    className="input-group"
                              >
                                    <div
                                          className="input-group-addon"
                                    >
                                          <i
                                                className="fa fa-user"
                                          />
                                    </div>
                                    <input
                                          className="form-control"
                                          name="firstName"
                                          onChange={[Function]}
                                          placeholder="First Name"
                                          type="text"
                                          value={undefined}
                                    />
                              </div>
                        </div>
                        <div
                              className="form-group"
                        >
                              <div
                                    className="input-group"
                              >
                                    <div
                                          className="input-group-addon"
                                    >
                                          <i
                                                className="fa fa-user"
                                          />
                                    </div>
                                    <input
                                          className="form-control"
                                          name="lastName"
                                          onChange={[Function]}
                                          placeholder="Last Name"
                                          type="text"
                                          value={undefined}
                                    />
                              </div>
                        </div>
                        <div
                              className="form-group"
                        >
                              <div
                                    className="input-group"
                              >
                                    <div
                                          className="input-group-addon"
                                    >
                                          <i
                                                className="fa fa-envelope"
                                          />
                                    </div>
                                    <input
                                          className="form-control"
                                          name="email"
                                          onChange={[Function]}
                                          placeholder="Email"
                                          type="text"
                                          value=""
                                    />
                              </div>
                        </div>
                        <div
                              className="form-group"
                        >
                              <div
                                    className="input-group"
                              >
                                    <div
                                          className="input-group-addon"
                                    >
                                          <i
                                                className="fa fa-lock"
                                          />
                                    </div>
                                    <input
                                          className="form-control"
                                          name="password"
                                          onChange={[Function]}
                                          placeholder="Password"
                                          type="password"
                                          value=""
                                    />
                              </div>
                        </div>
                        <div
                              className="buttonGroup"
                        >
                              <div
                                    className="form-group buttonGroup"
                              >
                                    <input
                                          className="signup btn btn-primary"
                                          type="submit"
                                          value="Signup"
                                    />
                              </div>
                        </div>
                  </form>
                  <div>
                        <p
                              className="signupRedirect"
                        >
                              <Link
                                    replace={false}
                                    to="/login"
                              >
                                    Have an account? Login Here
                              </Link>
                        </p>
                  </div>
                  <div />
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Signup
        email=""
        error=""
        handleChange={[Function]}
        handleSubmit={[Function]}
        password=""
/>,
      "_debugID": 40,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "email": "",
          "error": "",
          "handleChange": [Function],
          "handleSubmit": [Function],
          "password": "",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="container "
>
          <div
                    className="centeredContainer"
          >
                    <div
                              className="borderBox "
                    >
                              <div
                                        className="loginFormHeader"
                              >
                                        <div
                                                  className="loginIcon"
                                        >
                                                  <i
                                                            className="fa fa-shield fa-5x"
                                                  />
                                        </div>
                              </div>
                              <form
                                        id="loginform"
                                        onSubmit={[Function]}
                              >
                                        <div
                                                  className="form-group"
                                        >
                                                  <div
                                                            className="input-group"
                                                  >
                                                            <div
                                                                      className="input-group-addon"
                                                            >
                                                                      <i
                                                                                className="fa fa-user"
                                                                      />
                                                            </div>
                                                            <input
                                                                      className="form-control"
                                                                      name="firstName"
                                                                      onChange={[Function]}
                                                                      placeholder="First Name"
                                                                      type="text"
                                                                      value={undefined}
                                                            />
                                                  </div>
                                        </div>
                                        <div
                                                  className="form-group"
                                        >
                                                  <div
                                                            className="input-group"
                                                  >
                                                            <div
                                                                      className="input-group-addon"
                                                            >
                                                                      <i
                                                                                className="fa fa-user"
                                                                      />
                                                            </div>
                                                            <input
                                                                      className="form-control"
                                                                      name="lastName"
                                                                      onChange={[Function]}
                                                                      placeholder="Last Name"
                                                                      type="text"
                                                                      value={undefined}
                                                            />
                                                  </div>
                                        </div>
                                        <div
                                                  className="form-group"
                                        >
                                                  <div
                                                            className="input-group"
                                                  >
                                                            <div
                                                                      className="input-group-addon"
                                                            >
                                                                      <i
                                                                                className="fa fa-envelope"
                                                                      />
                                                            </div>
                                                            <input
                                                                      className="form-control"
                                                                      name="email"
                                                                      onChange={[Function]}
                                                                      placeholder="Email"
                                                                      type="text"
                                                                      value=""
                                                            />
                                                  </div>
                                        </div>
                                        <div
                                                  className="form-group"
                                        >
                                                  <div
                                                            className="input-group"
                                                  >
                                                            <div
                                                                      className="input-group-addon"
                                                            >
                                                                      <i
                                                                                className="fa fa-lock"
                                                                      />
                                                            </div>
                                                            <input
                                                                      className="form-control"
                                                                      name="password"
                                                                      onChange={[Function]}
                                                                      placeholder="Password"
                                                                      type="password"
                                                                      value=""
                                                            />
                                                  </div>
                                        </div>
                                        <div
                                                  className="buttonGroup"
                                        >
                                                  <div
                                                            className="form-group buttonGroup"
                                                  >
                                                            <input
                                                                      className="signup btn btn-primary"
                                                                      type="submit"
                                                                      value="Signup"
                                                            />
                                                  </div>
                                        </div>
                              </form>
                              <div>
                                        <p
                                                  className="signupRedirect"
                                        >
                                                  <Link
                                                            replace={false}
                                                            to="/login"
                                                  >
                                                            Have an account? Login Here
                                                  </Link>
                                        </p>
                              </div>
                              <div />
                    </div>
          </div>
</div>,
        "_debugID": 41,
        "_renderedOutput": <div
          className="container "
>
          <div
                    className="centeredContainer"
          >
                    <div
                              className="borderBox "
                    >
                              <div
                                        className="loginFormHeader"
                              >
                                        <div
                                                  className="loginIcon"
                                        >
                                                  <i
                                                            className="fa fa-shield fa-5x"
                                                  />
                                        </div>
                              </div>
                              <form
                                        id="loginform"
                                        onSubmit={[Function]}
                              >
                                        <div
                                                  className="form-group"
                                        >
                                                  <div
                                                            className="input-group"
                                                  >
                                                            <div
                                                                      className="input-group-addon"
                                                            >
                                                                      <i
                                                                                className="fa fa-user"
                                                                      />
                                                            </div>
                                                            <input
                                                                      className="form-control"
                                                                      name="firstName"
                                                                      onChange={[Function]}
                                                                      placeholder="First Name"
                                                                      type="text"
                                                                      value={undefined}
                                                            />
                                                  </div>
                                        </div>
                                        <div
                                                  className="form-group"
                                        >
                                                  <div
                                                            className="input-group"
                                                  >
                                                            <div
                                                                      className="input-group-addon"
                                                            >
                                                                      <i
                                                                                className="fa fa-user"
                                                                      />
                                                            </div>
                                                            <input
                                                                      className="form-control"
                                                                      name="lastName"
                                                                      onChange={[Function]}
                                                                      placeholder="Last Name"
                                                                      type="text"
                                                                      value={undefined}
                                                            />
                                                  </div>
                                        </div>
                                        <div
                                                  className="form-group"
                                        >
                                                  <div
                                                            className="input-group"
                                                  >
                                                            <div
                                                                      className="input-group-addon"
                                                            >
                                                                      <i
                                                                                className="fa fa-envelope"
                                                                      />
                                                            </div>
                                                            <input
                                                                      className="form-control"
                                                                      name="email"
                                                                      onChange={[Function]}
                                                                      placeholder="Email"
                                                                      type="text"
                                                                      value=""
                                                            />
                                                  </div>
                                        </div>
                                        <div
                                                  className="form-group"
                                        >
                                                  <div
                                                            className="input-group"
                                                  >
                                                            <div
                                                                      className="input-group-addon"
                                                            >
                                                                      <i
                                                                                className="fa fa-lock"
                                                                      />
                                                            </div>
                                                            <input
                                                                      className="form-control"
                                                                      name="password"
                                                                      onChange={[Function]}
                                                                      placeholder="Password"
                                                                      type="password"
                                                                      value=""
                                                            />
                                                  </div>
                                        </div>
                                        <div
                                                  className="buttonGroup"
                                        >
                                                  <div
                                                            className="form-group buttonGroup"
                                                  >
                                                            <input
                                                                      className="signup btn btn-primary"
                                                                      type="submit"
                                                                      value="Signup"
                                                            />
                                                  </div>
                                        </div>
                              </form>
                              <div>
                                        <p
                                                  className="signupRedirect"
                                        >
                                                  <Link
                                                            replace={false}
                                                            to="/login"
                                                  >
                                                            Have an account? Login Here
                                                  </Link>
                                        </p>
                              </div>
                              <div />
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Signup
    email=""
    error=""
    handleChange={[Function]}
    handleSubmit={[Function]}
    password=""
/>,
}
`;
